crates/oxc_linter/src/rules/eslint/no_extra_boolean_cast.rs

```diff
@@ -0,0 +1,879 @@
+        AstKind::LogicalExpression(expr) => {
+            (expr.operator == LogicalOperator::And || expr.operator == LogicalOperator::Or)
+                && enforce_for_logical_operands
+        }
+        _ => false,
+    }
+}
+
+fn is_bool_fn_or_constructor_call(node: &AstNode) -> bool {
+    match node.kind() {
+        AstKind::CallExpression(expr) => {
+            let x = expr.callee.without_parenthesized().get_identifier_reference().unwrap();
+            x.name == "Boolean"
+        }
+        AstKind::NewExpression(expr) => {
+            let x = expr.callee.without_parenthesized().get_identifier_reference().unwrap();
+            x.name == "Boolean"
+        }
+        _ => false,
+    }
+}
+
+fn is_first_arg(node: &AstNode, parent: &AstNode) -> bool {
+    match parent.kind() {
+        AstKind::CallExpression(expr) => expr.arguments.first().map_or(false, |arg| {
+            if let Argument::Expression(expr) = arg {
+                expr.without_parenthesized().span() == node.kind().span()
+            } else {
+                false
+            }
+        }),
+        AstKind::NewExpression(expr) => expr.arguments.first().map_or(false, |arg| {
+            if let Argument::Expression(expr) = arg {
+                expr.without_parenthesized().span() == node.kind().span()
+            } else {
+                false
+            }
+        }),
+        _ => false,
+    }
+}
+
+fn is_inside_test_condition(node: &AstNode, ctx: &LintContext) -> bool {
+    get_real_parent(node, ctx).map_or(false, |parent| match parent.kind() {
+        AstKind::IfStatement(stmt) => {
+            let expr_span = stmt.test.get_inner_expression().without_parenthesized().span();
+            expr_span == node.kind().span()
+        }
+        AstKind::DoWhileStatement(stmt) => {
+            let expr_span = stmt.test.get_inner_expression().without_parenthesized().span();
+            expr_span == node.kind().span()
+        }
+        AstKind::WhileStatement(stmt) => {
+            let expr_span = stmt.test.get_inner_expression().without_parenthesized().span();
+            expr_span == node.kind().span()
+        }
+        AstKind::ConditionalExpression(stmt) => {
+            let expr_span = stmt.test.get_inner_expression().without_parenthesized().span();
+            expr_span == node.kind().span()
+        }
+        AstKind::ForStatement(stmt) => stmt.test.as_ref().map_or(false, |expr| {
+            let expr_span = expr.get_inner_expression().without_parenthesized().span();
+            expr_span == node.kind().span()
+        }),
+        _ => false,
+    })
+}
+
+fn is_unary_negation(node: &AstNode) -> bool {
+    match node.kind() {
+        AstKind::UnaryExpression(expr) => expr.operator == UnaryOperator::LogicalNot,
+        _ => false,
+    }
+}
+
+fn get_real_parent<'a, 'b>(node: &AstNode, ctx: &'a LintContext<'b>) -> Option<&'a AstNode<'b>> {
+    let mut parent = ctx.nodes().parent_node(node.id());
+
+    if parent.is_some() {
```

Comment:
consider using `if let Some(parent) = parent {`, then using the unpacked `parent` below instead of using `unwrap()`